api/src/main/java/uk/gov/dhsc/htbhf/claimant/message/processor/DetermineEntitlementMessageProcessor.java

```diff
@@ -59,16 +67,22 @@ public MessageType supportsMessageType() {
     public MessageStatus processMessage(Message message) {
 
         DetermineEntitlementMessageContext messageContext = messageContextLoader.loadDetermineEntitlementContext(message);
-        Claimant claimant = messageContext.getClaim().getClaimant();
+        Claim claim = messageContext.getClaim();
+        Claimant claimant = claim.getClaimant();
         PaymentCycle currentPaymentCycle = messageContext.getCurrentPaymentCycle();
         PaymentCycle previousPaymentCycle = messageContext.getPreviousPaymentCycle();
 
         EligibilityResponse eligibilityResponse = eligibilityService.determineEligibility(claimant);
-        PaymentCycleVoucherEntitlement voucherEntitlement = determineVoucherEntitlement(
-                currentPaymentCycle,
-                previousPaymentCycle,
-                eligibilityResponse,
-                claimant);
+        PaymentCycleVoucherEntitlement voucherEntitlement = null;
+        if (EligibilityStatus.ELIGIBLE == eligibilityResponse.getEligibilityStatus()) {
```

Comment:
could we move the contents of the `if` statement into a separate method?

Reply:
Yes, this does need a bit of a refactor.